docs/clion_thin_client.md

```diff
@@ -0,0 +1,187 @@
+ssh-keygen -t ed25519 -b 4096
+```
+
+When prompted, choose a filename for your keys and leave the passphrase field
+empty.  
+
+Next, copy the public key you just created to the remote server. 
+```bash
+ssh-copy-id -i ~/.ssh/<keyname> <computingID>@portal.cs.virginia.edu
+```
+
+When prompted, provide your UVA CS password.  
+
+Once complete you should be able to ssh to the portal cluster without a password.  
+```bash
+ssh -i ~/.ssh/<keyname> <computingID>@portal.cs.virginia.edu
+```
+
+You can simplify the login experience with an [`ssh` config file][2]. In the
+`~/.ssh` directory create a file named `config`, and add an entry that looks
+like,
+```
+Host portal
+  User <computingID>
+  HostName portal.cs.virginia.edu
+  IdentityFile ~/.ssh/uva/<keyname>
+```
+
+Now, you can login with `ssh portal`.
+
+## Cloning tipc
+From `portal`, clone `tipc`.
+
+```bash
+git clone https://github.com/matthewbdwyer/tipc.git
+```
+
+## Configuring the Remote environment.
+You can configure your environment on `portal` by loading and unloading
+[modules][3]. The `tipc` project includes [modulefiles][4] to help configure
+the environment. The modulefiles are in the `conf` directory.  
+
+You can load the tipc module with the following command.
+```bash
+module load ~/tipc/conf/modulefiles/tipc/F23
+```
+
+Please note that there is an assumption in the stock modulefile that `tipc` is
+cloned to your home directory. If that assumption does not hold true, update
+the `topdir` variable in `conf/modules/tipc/F23` to the actual location of the
+`tipc` repository.  
+
+To confirm the module was successfully loaded check the value of the `TIPCLANG`
+environment variable.  
+```bash
+echo $TIPCLANG
+# /sw/ubuntu-22.04/llvm/14.0.6/bin/clang-14
+```
+
+To load the `tipc` environment every time you ssh to portal, add the `module
+load` command to your `bashrc`.
+```bash
+echo 'module load ~/tipc/conf/modulefiles/tipc/F23' >> ~/.bashrc
+```
+
+## Building tipc from the Command Line.
+At this point you should be fully equipped to build tipc on portal from the command line.
+
+From the `tipc` project directory,
+```bash
+mkdir build
+cd build
+cmake ..
+make -j4
+```
+
+## Using the JetBrains thin client
+You may prefer to develop with an IDE. In that case you can use the [JetBrains
+Gateway and thin client][6]. In this setup the IDE runs entirely on the server.
+You connect to the IDE from a thin client on your local machine over SSH.  
+
+[JetBrains Gateway][5] can be installed as a standalone app. It also comes part
+of any JetBrains IDE distribution. That is, if you have CLion installed
+already, you need not install the gateway. The instructions will continue with
+the JetBrains Gateway standalone app but are nearly identical, and have been
+tested with CLion.  
+
+First, launch the JetBrains Gateway application. From the "All Providers" page
+select "New Connection" under "SSH Connection".
+
+\image html gateway_all_provider_new_connection.png
```

Comment:
This is not showing up in GitHub as an image, maybe try - `![](assets/gateway_all_provider_new_connection.png)`

This is the case with all \image tags

Reply:
Otherwise looks good

Reply:
oh thats a doxygen tag, I wasn't thinking about viewing from gh, good catch. Fixed.